module/VuFind/src/VuFind/Recommend/EDSDatabases.php

```diff
@@ -0,0 +1,174 @@
+     * @var LibGuides
+     */
+    protected $libGuides;
+
+    /**
+     * Constructor
+     *
+     * @param LibGuides    $libGuides LibGuides API connection
+     * @param Config       $config    LibGuides API configuration object
+     * @param CacheAdapter $cache     Object cache
+     */
+    public function __construct(
+        LibGuides $libGuides,
+        Config $config,
+        CacheAdapter $cache
+    ) {
+        $this->libGuides = $libGuides;
+        $this->setCacheStorage($cache);
+
+        // Cache the data related to profiles for up to 10 minutes:
+        $this->cacheLifetime = intval($config->GetAZ->cache_lifetime ?? 600);
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called before the Search Results object performs its main search
+     * (specifically, in response to \VuFind\Search\SearchRunner::EVENT_CONFIGURED).
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function init($params, $request)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called after the Search Results object has performed its main search. This
+     * may be used to extract necessary information from the Search Results object
+     * or to perform completely unrelated processing.
+     *
+     * @param \VuFind\Search\Base\Results $results Search results object
+     *
+     * @return void
+     */
+    public function process($results)
+    {
+        $this->results = $results;
+    }
+
+    /**
+     * Get terms related to the query.
+     *
+     * @return array
+     */
+    public function getResults()
+    {
+        $records = $this->results->getResults();
+        $nameToDatabase = $this->getLibGuidesDatabases();
+        $databases = [];
+        foreach ($records as $record) {
+            $databaseInfo = $nameToDatabase[$record->getDbLabel()] ?? null;
+            if ($databaseInfo) {
+                $databases[$record->getDbLabel()] = $databaseInfo;
+            }
+        }
+        return $databases;
+    }
+
+    /**
+     * Load or retrieve from the cache the list of LibGuides A-Z databases.
+     *
+     * @return array An array mapping a database name to the full object
+     * from the LibGuides /az API.
+     */
+    protected function getLibGuidesDatabases()
+    {
+        $nameToDatabase = $this->getCachedData('libGuidesAZ-nameToDatabase');
```

Comment:
> Seems like it would be pretty easy to fall back to an .ini file if LibGuides is unsupported. Maybe it's worth considering adding some generic functionality of that sort to make this more broadly useful for institutions not using LibGuides.

I liked this idea; implemented.  The .ini configuration either replaces or overrides LibGuides's list.  There may though be better ways to define that configuration, I don't love url[] and open to ideas.  I am thinking in the future there may be other metadata we would also want to configure about each database via EDS.ini, matching some of the other info coming from LibGuides, if it was useful in the template.